File: src/main/java/pe/nanamochi/banchus/packets/server/MatchUpdatePacket.java
package pe.nanamochi.banchus.packets.server;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.entities.packets.Match;
import pe.nanamochi.banchus.packets.Packet;
import pe.nanamochi.banchus.packets.Packets;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class MatchUpdatePacket implements Packet {
  private Match match;
  private boolean shouldSendPassword;

  @Override
  public Packets getPacketType() {
    return Packets.BANCHO_MATCH_UPDATE;
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/server/MatchUpdatePacket.kt
package pe.nanamochi.banchus.packets.server

import java.io.OutputStream
import pe.nanamochi.banchus.components.Match
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.core.ServerPacket
import pe.nanamochi.banchus.io.DataWriter

class MatchUpdatePacket(var match: Match? = null, var shouldSendPassword: Boolean = false) :
    ServerPacket() {
    override val type = PacketType.BANCHO_MATCH_UPDATE

    override fun write(writer: DataWriter, out: OutputStream) {
        match?.write(writer, out, shouldSendPassword)
    }
}
